File: Minimum_Cost_to_Move_Chips_to_The_Same_Position.py
position = [1,2,3]

def minCostToMoveChips(position):
    even,odd = 0,0
    for pos in position:
        if pos%2==0:
            even+=1
        else:
            odd+=1
    
    return min(even,odd)

print(minCostToMoveChips(position))

File: Minimum_Cost_to_Move_Chips_to_The_Same_Position/Minimum_Cost_to_Move_Chips_to_The_Same_Position.go
package main

import "fmt"

func main() {
	position := []int{1, 2, 3}
	fmt.Println(minCostToMoveChips(position))
}

func minCostToMoveChips(position []int) int {
	even := 0
	odd := 0
	for _, pos := range position {
		if pos%2 == 0 {
			even += 1
		} else {
			odd += 1
		}
	}

	if even > odd {
		return odd
	} else {
		return even
	}
}
